### PR TITLE
Color for vtx ready status

### PR DESCRIPTION
**Pull request by HThuren**
Change to Device ready has green / red color, are moved up in presentation and are updated if device status change.
image

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Review by McGiverGim — CHANGES_REQUESTED** [review]
It's cleaner for the translators simply apply the class to the element depending on the content. It's how we it usually in the rest of the code and does not imply new messages with class included.

**Comment by HThuren** [comment]
> It's cleaner for the translators simply apply the class to the element depending on the content. It's how we it usually in the rest of the code and does not imply new messages with class included.

I am not sure what you address, I used the same setup as GPS fixed, ?

**Comment by McGiverGim** [comment]
Sorry for my late response, but I was out. What I was talking about is to do something like this:

https://github.com/betaflight/betaflight-configurator/blob/9d9edf9266019a0a84d667ed649f64fa5bda73ab/src/js/serial_backend.js#L681

More info here: http://api.jquery.com/toggleclass/

This will add/remove the class depending on the second parameter true/false. In this way the messages file is clean, only with basic 'true' and 'false', and we use this in the same part of the code where we decide what to show, and where we assign MSP values to fields.

In the CSS file you can have a "red" value for the 'basic' class, and a gree value when the 'basic' class has a 'valid' class too, something similar to (the names are only for reference, better to use something more significative):
```
.basic {
  color: red;
}
.basic .valid {
  color: green;
}
```

Add a "custom" method like your latest version, I think is not the best way, because maybe we want it for YES/NO, or TRUE/FALSE, or ACEPTED/REJECTED, or VALID/NOT VALID... etc.

**Comment by HThuren** [comment]
@McGiverGim , no worry, I also have a fulltime job :-)

Very smart and rigorous way you decribe, I go for it .-.

**Comment by HThuren** [comment]
hi @McGiverGim 

Well, I havn't suceed in the requested change, don't you think we still have a 'clean' message file with this construction

```
export function getColorYesNo(value) {
    return (value ? `<span class="fixtrue">${i18n.getMessage("Yes")}</span>` : `<span class="fixfalse">${i18n.getMessage("No")}</span>`);
}
```

called fx
      `$('.GPS_info td.fix').html(getColorYesNo(FC.GPS_DATA.fix));`

**Comment by Benky** [comment]
Could it be something like this: https://github.com/Benky/betaflight-configurator/commit/aeaf2fd919e850f17b6ea8dd7b75d641c7cddcfc ?

**Comment by HThuren** [comment]
> Could it be something like this: [Benky@aeaf2fd](https://github.com/Benky/betaflight-configurator/commit/aeaf2fd919e850f17b6ea8dd7b75d641c7cddcfc) ?

So simple and elegant :-) - thats the way

**Comment by HThuren** [comment]
> Also notice plugging LIPO updates status - but removing LIPO the status remains. But is outside the scope of this PR.

yes, I'm aware, another PR. :-)

File: src/js/tabs/gps.js
Comment (haslinghuis) [inline src/js/tabs/gps.js:195]: Parenthesis not needed
 
```suggestion
            $('.GPS_info span.fix').text(FC.GPS_DATA.fix ? i18n.getMessage('gpsFixTrue') : i18n.getMessage('gpsFixFalse'));
```

File: src/js/tabs/gps.js
Comment (haslinghuis) [inline src/js/tabs/gps.js:196]:
```suggestion
            $('.GPS_info span.fix').toggleClass('active', FC.GPS_DATA.fix !== 0);
```

File: src/js/tabs/setup.js
Comment (haslinghuis) [inline src/js/tabs/setup.js:431]: same as above.
```suggestion
            gpsFix_e.text(FC.GPS_DATA.fix ? i18n.getMessage('gpsFixTrue') : i18n.getMessage('gpsFixFalse'));
```

File: src/js/tabs/setup.js
Comment (haslinghuis) [inline src/js/tabs/setup.js:432]:
```suggestion
            gpsFix_e.toggleClass("active", FC.GPS_DATA.fix !== 0);
```

File: src/js/tabs/vtx.js
Comment (haslinghuis) [inline src/js/tabs/vtx.js:303]:
```suggestion
        vtxReady_e.text(FC.VTX_CONFIG.vtx_device_ready ? i18n.getMessage('vtxReadyTrue') : i18n.getMessage('vtxReadyFalse'));
```

File: src/js/tabs/setup.js
Comment (HThuren) [reply src/js/tabs/setup.js:432]: Right, sharp eyes :-)

File: src/js/tabs/gps.js
Comment (HThuren) [reply src/js/tabs/gps.js:196]: Also ind setup.js

File: src/js/tabs/vtx.js
Comment (haslinghuis) [inline src/js/tabs/vtx.js:53]:
```suggestion
        vtxReady_e.text(FC.VTX_CONFIG.vtx_device_ready ? i18n.getMessage('vtxReadyTrue') : i18n.getMessage('vtxReadyFalse'));
```

**Review by haslinghuis — CHANGES_REQUESTED** [review]
One left :)

**Review by haslinghuis — APPROVED** [review]

**Comment by haslinghuis** [comment]
@McGiverGim should be good to go now

**Review by McGiverGim — CHANGES_REQUESTED** [review]
A lot more functionality very interesting. The code seems ok to me. 
But the less/css code can be a lot better using the less possibilities. Only one difference between the active and the basis class.
I'm on mobile, tomorrow I can suggest the exact code. Sorry but my real work has all my time lately.

**Comment by HThuren** [comment]
> A lot more functionality very interesting. The code seems ok to me. But the less/css code can be a lot better using the less possibilities. Only one difference between the active and the basis class. I'm on mobile, tomorrow I can suggest the exact code. Sorry but my real work has all my time lately.

No worry and hurry, I look forward to learn about CSS/less

File: src/css/main.less
Comment (McGiverGim) [inline src/css/main.less:1744]: I've not tested it, but what I wanted to say is to use something like this:
```suggestion
.fix {
    background-color: #e60000;
    padding: 2px 5px;
    border-radius: 3px;
    color: #fff;
    font-size: 10px;

    &.active {
        background-color: #56ac1d;
    }
}
.vtx_device_ready {
    background-color: #e60000;
    padding: 2px 5px;
    border-radius: 3px;
    color: #fff;
    font-size: 10px;
    
    &.active {
        background-color: #56ac1d;
    }
}
```
But, two suggestions more:
- Don't call `active` the css class, it can be confuse because the pseudo-class `:active` exists. Maybe something like `positive` is better.
- The `fix` and `vtx_device_ready` class are exactly the same. Why don't we use one for both? Being in the `main.less` class, something generic that can be used at each place where we want to use it.

**Review by McGiverGim — CHANGES_REQUESTED** [review]
Hi! Sorry for my late response, but I think this will work.

File: src/css/main.less
Comment (HThuren) [reply src/css/main.less:1744]: good suggestion, changed to this
```
.colorToggle {
    background-color: #e60000;
    padding: 2px 5px;
    border-radius: 3px;
    color: #fff;
    font-size: 10px;
	&.ready {
		background-color: #56ac1d;
	}
}
```
and
```
        const vtxReady_e = $('.VTX_info span.colorToggle');
        vtxReady_e.text((FC.VTX_CONFIG.vtx_device_ready) ? i18n.getMessage('vtxReadyTrue') : i18n.getMessage('vtxReadyFalse'));
        vtxReady_e.toggleClass('ready', FC.VTX_CONFIG.vtx_device_ready);
```

File: src/js/tabs/vtx.js
Comment (haslinghuis) [reply src/js/tabs/vtx.js:53]: This is still unresolved :stuck_out_tongue:

File: src/js/tabs/vtx.js
Comment (haslinghuis) [reply src/js/tabs/vtx.js:303]: Same

File: src/js/tabs/vtx.js
Comment (HThuren) [reply src/js/tabs/vtx.js:53]: argh

**Comment by haslinghuis** [comment]
@HThuren LGTM, only two unresolved issues remaining

**Review by McGiverGim — APPROVED** [review]
For the rest to me is ok. Thanks for the changes.

**Pull request merged**
